### PR TITLE
Added ros build script

### PR DESCRIPTION
**Pull request by UlysseVautier**
Test ros-bare build script (no automatic deb push at the moment)

### CODE REVIEW

File: docker/ros-bare.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/bin/sh
+
+set -o errexit
+set -o verbose
+
+echo "deb http://packages.ros.org/ros/ubuntu stretch main" > /etc/apt/sources.list.d/ros-latest.list && 
+apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-key 421C365BD9FF1F717815A3895523BAEEB01FA116
+apt update
+apt upgrade
```
Comment (Guillaumebeuzeboc) [inline docker/ros-bare.sh]: This upgrade seems to be useless

File: docker/ros-bare.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#!/bin/sh
+
+set -o errexit
+set -o verbose
+
+echo "deb http://packages.ros.org/ros/ubuntu stretch main" > /etc/apt/sources.list.d/ros-latest.list && 
+apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-key 421C365BD9FF1F717815A3895523BAEEB01FA116
+apt update
+apt upgrade 
+apt-get install build-essential python-yaml cmake subversion wget python-setuptools mercurial git-core libapr1-dev libaprutil1-dev libbz2-dev python-dev libgtest-dev python-paramiko libboost-all-dev liblog4cxx10-dev pkg-config python-empy swig python-nose lsb-release python-pip python-gtk2
```
Comment (Guillaumebeuzeboc) [inline docker/ros-bare.sh]: These dependencies are needed to compile ROS but do not depend on ROS. So these should be installed in the Dockerfile.

Maybe we could move Dockerfile install dependencies to a file, but I am not sure how Dockerhub would behave.

But we could also move the Docker build to travisCI then push the image to dockerhub

File: docker/ros-bare.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+pip install rosdep rosinstall_generator wstool rosinstall
```
Comment (Guillaumebeuzeboc) [inline docker/ros-bare.sh]: Same as above

File: docker/ros-bare.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+rosdep init 
+rosdep update
+cd /var/tmp/
+git clone https://github.com/Rob-Orb/Ros.git
+cd Ros
+mkdir catkin_ws 
+cd catkin_ws
```
Comment (Guillaumebeuzeboc) [inline docker/ros-bare.sh]: I would do:
mkdir -p Ros/catkin_ws
cd Ros/catkin_ws

File: docker/ros-bare.sh [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+cd Ros
+mkdir catkin_ws 
+cd catkin_ws 
+rosinstall_generator ros_comm --rosdistro kinetic --deps --wet-only --exclude roslisp --tar > kinetic-ros_comm-wet.rosinstall 
+wstool init src kinetic-ros_comm-wet.rosinstall 
+rosdep install --from-paths src --ignore-src --rosdistro kinetic -y -r --os=debian:stretch 
+cd src/catkin/cmake/test 
+sed -i "s/set_target_properties(${target} PROPERTIES EXCLUDE_FROM_ALL TRUE)/#set_target_properties(${target} PROPERTIES EXCLUDE_FROM_ALL TRUE)/g" gtest.cmake
+sed -i "s/set_target_properties(gtest gtest_main PROPERTIES EXCLUDE_FROM_ALL 1)/#set_target_properties(gtest gtest_main PROPERTIES EXCLUDE_FROM_ALL 1)/g" gtest.cmake
+cd ../../../../
```
Comment (Guillaumebeuzeboc) [inline docker/ros-bare.sh]: better to use pushd, popd

**Review by Guillaumebeuzeboc — COMMENTED** [review]
You can also fixup the second commit into the first one before merging

File: docker/ros-bare.sh (hunk 1 above)
Comment (UlysseVautier) [reply docker/ros-bare.sh]: Yes it is

File: docker/ros-bare.sh (hunk 4 above)
Comment (UlysseVautier) [reply docker/ros-bare.sh]: Yes

File: docker/ros-bare.sh (hunk 5 above)
Comment (UlysseVautier) [reply docker/ros-bare.sh]: Changed to simply cd -

File: docker/ros-bare.sh (hunk 2 above)
Comment (UlysseVautier) [reply docker/ros-bare.sh]: If in Dockerfile, it might be unnecessary for other builds. But might be faster for the Ros build.

Comment (Guillaumebeuzeboc) [reply docker/ros-bare.sh]: Yes but that's not a big deal. It's just gonna increase a little bit the size of the image.
And yes it's gonna make ROS build faster and move us a bit away from the 50 mins timeout

**Pull request merged**